great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py

```diff
@@ -109,6 +120,8 @@ class DataAssistantResult(SerializableDictDot):
         "column.mean": "expect_column_mean_to_be_between",
         "column.median": "expect_column_median_to_be_between",
         "column.standard_deviation": "expect_column_stdev_to_be_between",
+        "column.quantile_values": "expect_column_quantile_values_to_be_between",
+        ("column.min", "column.max"): "expect_column_values_to_be_between",
```

Comment:
> Could you add a type annotation to METRIC_EXPECTATION_MAP?
Yes, I figured out how! We couldn't figure it out before because this is a `dataclass`, but I found a solution using `default_factory`.

Reply:
> It seems like you don't want to interact with METRIC_EXPECTATION_MAP directly though but through some helper method like _get_metric_expectation_map so maybe that method should be public (ie remove the _) and this map should be private (add a _). Another option would be to make a new class with takes a dict in the initializer and then abstracts away this asymmetry in the keys. I guess this is a long winded way of saying, it would be good not to highlight this asymmetry in a public class variable and we should encapsulate it either by making it private and adding public methods in the class to interact with it or by moving it into it's own class.

The private/public methods were scrutinized pretty heavily by Abe and the team, so I am hesitant to make methods public that are only being used within `DataAssistantResult` objects and have no use in the public API. I'm leaning towards making the left hand side only tuples if we feel like this is adding too much complexity. I'm not following what we would gain by adding a new class because the contributors/users would still define this map the same way no?

Reply:
> To record one last thing we talked about. It is fine the right hand side are strings for now but if more than 1 expectation is derived from a metric they may become lists or tuples in the future.

I see this now, thank you!

Reply:
I didn't realize this was so heavily scrutinized. 

My main concern is this adds a new type to the keys (before it was just strings). I don't know where this dictionary is consumed but I'm worried that doing this puts a burden on the user. Are they going to now have to introduce a conditional to check types or do some other processing to prepare the key before doing a lookup? If so, we could centralize it in some accessor so it only appears once in the code and the user doesn't have to deal with it.

Reply:
The map will be defined by the user in their custom `DataAssistantResult` child wrapper class. It is needed for metrics and expectations that either do not have implemented plots, or are custom to their deployment. The idea is that they shouldn't need to override anything in the parent class that reads from this map. If they are using something already built by us (`OnboardingDataAssistant`) there is no need for them to even be aware of this variable.